datalad/distribution/publish.py

```diff
@@ -256,49 +282,91 @@ def __call__(path=None, dataset=None, to=None, recursive=False, since=None,
+
+            # publishing of `dest_resolved` might depend on publishing other
+            # remote(s) first:
+            # define config var name for potential publication dependencies
+            depvar = 'remote.{}.datalad-publish-depends'.format(dest_resolved)
+            for d in ds.config.get(depvar, []):
+                lgr.info("Dependency detected: '%s'" % d)
+                # Note: Additional info on publishing the dep. comes from within
+                # `ds.publish`.
+                ds.publish(path=path,
+                           to=d,
+                           since=since,
+                           skip_failing=skip_failing,
+                           recursive=recursive,
+                           recursion_limit=recursion_limit,
+                           git_opts=git_opts,
+                           annex_opts=annex_opts,
+                           annex_copy_opts=annex_copy_opts)
+
+            lgr.info("Publishing {0} to {1}".format(ds, dest_resolved))
+
+            # we now know where to push to:
+            # TODO: what to push? default: git push --mirror if nothing configured?
+            # consider also: --follow-tags, --tags, --atomic
+
+            # Note: git's push.default is 'matching', which possibly doesn't
+            # work for first
+            # time publication (a branch, that doesn't exist on remote yet)
+            # But if we want to respect remote.*.push entries, etc. we need to
+            # not pass a specific refspec (like active branch) to `git push`
+            # by default.
+
+            _log_push_info(ds.repo.push(remote=dest_resolved,
+                                        refspec=ds.repo.get_active_branch(),
+                                        set_upstream=set_upstream))
 
             published.append(ds)
 
             if publish_files or annex_copy_opts:
                 if not isinstance(ds.repo, AnnexRepo):
-                    raise RuntimeError(
-                        "Cannot publish content of something, that is not "
-                        "part of an annex. ({0})".format(ds))
+                    # incomplete, since `git push` was done already:
+                    raise IncompleteResultsError(
+                        (published, skipped),
+                        failed=publish_files,
+                        msg="Cannot publish content of something, that is not "
+                            "an annex. ({0})".format(ds))
+                if ds.config.get('remote.{}.annex-ignore', False):
+                    # Q: Do we need a --force option here? annex allows to
+                    # ignore the ignore setting
+                    raise IncompleteResultsError(
+                        (published, skipped),
+                        failed=publish_files,
+                        msg="Sibling '{0}' of {1} is configured to be ignored "
+                            "by annex. No content was published."
+                            % (dest_resolved, ds))
```

Comment:
This one too... May be I am not entirely clear on the flow and logic here - but in both cases wouldn't it interrupt publishing as soon as it encounters one of those cases?